### PR TITLE
refactor(router): renaming showView with showViewGenerator

### LINKED ISSUE #1603: router.js->showView: confusing name
@vladikoff pointed out that `router.js` contains two `showView` functions, and the first one does not actually show a view, it creates and returns a function that shows a view.  See https://github.com/mozilla/fxa-content-server/blob/f3352064f7e5acd7d5c2488287070c00c6f67737/app/scripts/router.js#L74-L87

### PR DESCRIPTION
**Pull request by ava14**
Fixes #1603

### CODE REVIEW

**Comment by vladikoff** [comment]
Thanks for the pull request!

Tests also need to be renamed: https://github.com/mozilla/fxa-content-server/blob/44bfc688dfbf96cedac8bf85f68795e4c12b1d16/app/tests/spec/lib/router.js#L145

**Comment by vladikoff** [comment]
Actually nevermind, that seems to be a different showView. Confusing indeed...

**Comment by vladikoff** [comment]
I guess this is the only line that is missing test coverage in this module:
image

**Comment by shane-tomlinson** [comment]
@ava14 - we made some other edits to router.js, can you rebase this PR?

**Comment by vladikoff** [comment]
ping @ava14

**Comment by ava14** [comment]
Hi @vladikoff & @shane-tomlinson, 
I am presently working on setting up the environment on my machine to fix this issue.

**Comment by vladikoff** [comment]
@ava14 great, see https://github.com/mozilla/fxa-local-dev

**Comment by ava14** [comment]
@vladikoff @riadhchtara @shane-tomlinson Hi guys, I tried to rebase this PR and created the test. Could you please review it.

**Comment by vladikoff** [comment]
```
 Total 11.1s
        Warning: Running "jscs:src" (jscs) task
    Expected indentation of 4 characters at app/tests/spec/lib/router.js :
       450 |    });
       451 |
       452 |   describe('showViewGenerator', function () {
    ------------^
       453 |      it('generates and shows view', function () {
       454 |        assert.isFunction(router.showViewGenerator(SignUpView, { canGoBack: false })());
    >> 1 code style errors found!
    Warning: Task "jscs:src" failed. Use --force to continue.
```

File: app/scripts/router.js [hunk 1]
```diff
@@ -284,6 +284,8 @@ function (
         });
     },
 
+    showViewGenerator: showViewGenerator,
```
Comment (vladikoff) [inline app/scripts/router.js]: @zaach  Should we add this or have a private ref: `_showViewGenerator` just for testing purposes?

**Comment by riadhchtara** [comment]
@ava14  you just need to fix the indentation issue and to be safe, rename the function as @vladikoff  has told you.
Thanks for your contribution.

File: app/scripts/router.js (hunk 1 above)
Comment (riadhchtara) [inline app/scripts/router.js]: @vladikoff : @ava14 was wondering today whether or not  you are going to do the merge, if he renames  the function?
thanks :)

**Comment by ava14** [comment]
@riadhchtara @vladikoff @shane-tomlinson  Hi all, I have fixed indent issue and renamed the function. Please see if it is ok :)

**Comment by vladikoff** [comment]
```
Failed test: generates and shows view
 => this.createAndShowView is not a function
```

**Comment by vladikoff** [comment]
Tests failing and needs to be rebased.

**Pull request closed without merging**